src/Tools/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/AnalysisExclusions.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.CodeAnalysis.Testing
+{
+    /// <summary>
+    /// Specifies characteristics of a file which cause diagnostics to not be reported.
+    /// </summary>
+    [Flags]
+    public enum AnalysisExclusions
+    {
+        /// <summary>
+        /// No special exclusions apply.
+        /// </summary>
+        None = 0,
+
+        /// <summary>
+        /// Analysis will not report diagnostics in generated code.
+        /// </summary>
+        GeneratedCode = 0x01,
+
+        /// <summary>
+        /// Diagnostics will not be reported if a <c>#pragma warning disable</c> appears at the beginning of the file.
+        /// </summary>
+        Suppression = 0x02,
```

Comment:
When is this ever *not* the case? Do analyzers even have the choice to show diagnostics in spite of such a `#pragma`? Or is the question whether the analyzer notices the `#pragma` and saves itself the effort of producing diagnostics that it figures Roslyn will filter out anyway?

Reply:
StyleCop Analyzers has a rule that currently fails this (bug linked in the initial PR comment). Providing this option allows StyleCop Analyzers to switch to the new testing framework and file the bug as a follow-up issue, rather than block adoption altogether.

Reply:
Also test inputs that contain a `#pragma warning enable` somewhere in the file might need to remove this flag, because it tests the case by adding the line to the top of the file.